aten/src/ATen/core/TensorImpl.h

```diff
@@ -1427,8 +1434,8 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
   at::Storage storage_; // TODO: Fix visibility on me
 
 protected:
-  std::vector<int64_t> sizes_;
-  std::unique_ptr<int64_t[]> strides_; // this saves two words
+  at::SmallVector<int64_t,3> sizes_;
```

Comment:
I know you said "don't review". I just want to say that I would use `4` or even `5` here because 4d tensors are very very common. :)